lib/Sema/CSApply.cpp

```diff
@@ -298,6 +298,7 @@ namespace {
     ConstraintSystem &cs;
     DeclContext *dc;
     Solution &solution;
+    Optional<SolutionApplicationTarget> target;
```

Comment:
This shouldn't be `Optional`, but there's a `Solution::coerceCallArguments` that creates an `ExprRewriter` just to call `ExprRewriter::coerceCallArguments`...

Reply:
We could probably refactor away `Solution::coerceToType` later. It looks unnecessary.